app/uk/gov/hmrc/helptosave/repo/EnrolmentStore.scala

```diff
@@ -87,76 +85,68 @@ object EnrolmentStore {
 }
 
 @Singleton
-class MongoEnrolmentStore @Inject() (mongo:   ReactiveMongoComponent,
+class MongoEnrolmentStore @Inject() (mongo:   MongoComponent,
                                      metrics: Metrics)(implicit ec: ExecutionContext)
-  extends ReactiveRepository[EnrolmentData, BSONObjectID](
+  extends PlayMongoRepository[EnrolmentData](
+    mongoComponent = mongo,
     collectionName = "enrolments",
-    mongo          = mongo.mongoConnector.db,
-    EnrolmentData.ninoFormat,
-    ReactiveMongoFormats.objectIdFormats)
-  with EnrolmentStore {
-
-  val log: Logger = new Logger(logger)
+    domainFormat   = EnrolmentData.ninoFormat,
+    indexes        = Seq(IndexModel(ascending("nino"), IndexOptions().name("ninoIndex").unique(false)))
+  )
+  with EnrolmentStore with Logging {
 
   def getRegex(nino: String): String = "^" + nino.take(8) + ".$"
 
-  override def indexes: Seq[Index] = Seq(
-    Index(
-      key  = Seq("nino" → IndexType.Ascending),
-      name = Some("ninoIndex")
-    )
-  )
-
   private[repo] def doInsert(nino:              NINO,
                              eligibilityReason: Option[Int],
                              source:            String,
                              itmpFlag:          Boolean,
-                             accountNumber:     Option[String])(implicit ec: ExecutionContext): Future[WriteResult] = {
+                             accountNumber:     Option[String])(implicit ec: ExecutionContext): Future[Unit] = {
     accountNumber match {
-      case Some(accountNum) ⇒ collection.insert(ordered = false).one(BSONDocument("nino" -> nino, "itmpHtSFlag" -> itmpFlag,
-        "eligibilityReason" -> eligibilityReason, "source" -> source, "accountNumber" -> accountNum))
+      case Some(accountNumber) ⇒
```

Comment:
I think we don't need pattern matching here
collection.insertOne(
          EnrolmentData(
            nino              = nino,
            itmpHtSFlag       = itmpFlag,
            eligibilityReason = eligibilityReason,
            source            = Some(source),
            accountNumber     = accountNumber
        ).toFuture().map(_ ⇒ ())
        This would also do as accountNumber is optional. If it has value , it will be inserted else None wil go. What do you think?

Reply:
resolved